db-tests/src/test/java/gov/nasa/jpl/aerie/database/ActivityDirectiveChangelogTests.java

```diff
@@ -0,0 +1,310 @@
+package gov.nasa.jpl.aerie.database;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ActivityDirectiveChangelogTests {
+
+  private static final File initSqlScriptFile = new File("../merlin-server/sql/merlin/init.sql");
+  private DatabaseTestHelper helper;
+  private MerlinDatabaseTestHelper merlinHelper;
+
+  private Connection connection;
+  int fileId;
+  int missionModelId;
+
+  @BeforeEach
+  void beforeEach() throws SQLException {
+    fileId = merlinHelper.insertFileUpload();
+    missionModelId = merlinHelper.insertMissionModel(fileId);
+  }
+
+  @AfterEach
+  void afterEach() throws SQLException {
+    helper.clearTable("uploaded_file");
+    helper.clearTable("mission_model");
+    helper.clearTable("plan");
+    helper.clearTable("activity_directive_changelog");
+    helper.clearTable("activity_directive");
+  }
+
+  @BeforeAll
+  void beforeAll() throws SQLException, IOException, InterruptedException {
+    helper = new DatabaseTestHelper(
+        "aerie_merlin_test",
+        "Merlin Database Tests",
+        initSqlScriptFile
+    );
+    helper.startDatabase();
+    connection = helper.connection();
+    merlinHelper = new MerlinDatabaseTestHelper(connection);
+  }
+
+  @AfterAll
+  void afterAll() throws SQLException, IOException, InterruptedException {
+    helper.stopDatabase();
+    connection = null;
+    helper = null;
+  }
+
+  @Test
+  void shouldHaveNoChangelogsForEmptyPlan() throws SQLException {
+    final var expectedCount = 0;
+
+    final var planId = merlinHelper.insertPlan(
+        missionModelId,
+        merlinHelper.admin.name(),
+        "changelog plan" + UUID.randomUUID(),
+        "1970-1-1 00:00:00+00");
+
+    final var res = connection.createStatement()
+                              .executeQuery(
+                                  // language=sql
+                                  """
+                                      SELECT count(revision)
+                                      FROM activity_directive_changelog
+                                      WHERE plan_id = %d;"""
+                                      .formatted(planId)
```

Comment:
(General Feedback for the file) Please use try-with-resources blocks instead.

```suggestion
try(final var statement = connection.createStatement()){
      final var res =  statement.executeQuery(
          """
          SELECT count(revision) FROM activity_directive_changelog
          WHERE plan_id = %d;
          """.formatted(planId)
      );
 }
```